File: plugins/swift-kit/testdata/good/Sources/Ok.swift
// Good fixture: optional bind, documented allow.

func firstOk(_ values: [String?]) -> String? {
    guard let first = values.first else { return nil }
    return first ?? "missing"
}

func tryOk(_ raw: String) throws -> Int {
    guard let value = Int(raw) else {
        throw NSError(domain: "fixture", code: 1)
    }
    return value
}

// Documented intentional seam (boot probe); keep allow on the smell line.
func documentedLegacy(_ value: String?) -> String {
    return value!  // swift-rg-allow: fixture documents allow marker for intentional force unwrap seam
}

File: plugins/kotlin-kit/testdata/good/src/main/kotlin/demo/Ok.kt
// Good fixture: optional bind, documented allow.

fun firstOk(values: List<String?>): String? {
    return values.firstOrNull() ?: "missing"
}

fun tryOk(raw: String): Int {
    return requireNotNull(raw.toIntOrNull()) { "bad int" }
}

// Documented intentional seam (boot probe); keep allow on the smell line.
fun documentedLegacy(value: String?): String {
    return value!! // kotlin-rg-allow: fixture documents allow marker for intentional force unwrap seam
}
